mne/cov.py

```diff
@@ -545,14 +554,45 @@ def _check_method_params(method, method_params, keep_sample_mean=True,
             if v.get('assume_centered', None) is False:
                 raise ValueError('`assume_centered` must be True'
                                  ' if `keep_sample_mean` is False')
-    return method, _method_params
+    return method, _method_params, rank
+
+
+def _check_rank(rank, method, was_auto=False):
+    if isinstance(rank, string_types):
```

Comment:
some simple doc string ?

Reply:
@larsoner Can you verbally describe what is the goal here. The moment where I see code from the `operator' module in use I start getting worried ...

Reply:
`operator.index` is just the preferred way to check if something is an int

Reply:
... and the `''` is just a way to deprecate `rank='full'` (current `master` behavior) to `rank=None` (auto-detect and preserve)